tests/test_cases/issue_843/issue_843.py

```diff
@@ -0,0 +1,40 @@
+import cocotb
+from cocotb.triggers import Event, Timer
+
+
+@cocotb.test()
+def test_yield_list_stale(dut):
+    """ Test that a trigger yielded as part of a list can't cause a spurious wakeup """
+    # gh-843
```

Comment:
This looks like it would fit well in `test_concurrency_primitives.py`

Reply:
It seems you favor not moving tests tied to issues into their own dirs. Do you think that other issue dirs should be absorbed into test_cocotb?

Reply:
I don't think you should increase work for yourself and do that as part of this PR, but in general, yes.

It's a good idea to _reference_ issues in the tests, but naming after the issue makes it impossible to tell when writing a new test whether a similar one already exists - especially now that you've spent a lot of effort categorizing the `test_cocotb` tests!

Only for an exceptionally contrived case would I recommend using an issue number.